### PR TITLE
FIX copy with versioning

### PR DESCRIPTION
**Pull request by nicolas2bert**

### CODE REVIEW

File: tests/functional/aws-node-sdk/test/versioning/objectCopy.js [hunk 1]
```diff
@@ -133,6 +134,45 @@ testing('Object Version Copy', () => {
             });
         }
 
+        it('should copy the appropriate version into the same bucket',
+        () => s3.putObjectAsync({ Bucket: sourceBucketName,
+              Key: sourceObjName,
+              Body: secondContent })
+            .then(() => s3.copyObjectAsync({
+                Bucket: sourceBucketName,
+                CopySource: `${sourceBucketName}/${sourceObjName}?` +
```
Comment (electrachong) [inline tests/functional/aws-node-sdk/test/versioning/objectCopy.js]: minor: it should be the same as the `copySource` variable set in the beforeEach, right?

File: tests/functional/aws-node-sdk/test/versioning/objectCopy.js [hunk 2]
```diff
@@ -133,6 +134,45 @@ testing('Object Version Copy', () => {
...
+            }));
+
+        it('should copy the appropriate version into a different ' +
+        'destination bucket',
+        () => s3.putObjectAsync({ Bucket: sourceBucketName,
+              Key: sourceObjName,
+              Body: secondContent })
+            .then(() => s3.copyObjectAsync({
+                Bucket: destBucketName,
+                CopySource: `${sourceBucketName}/${sourceObjName}?` +
```
Comment (electrachong) [inline tests/functional/aws-node-sdk/test/versioning/objectCopy.js]: same as above, it should be the same as the `copySource` variable set in the beforeEach, right?

File: tests/functional/aws-node-sdk/test/versioning/objectCopy.js [hunk 3]
```diff
@@ -617,7 +620,7 @@ testing('Object Version Copy', () => {
                     Bucket: destBucketName,
                     Key: destObjName,
                     CopySource: `${sourceBucketName}/${sourceObjName}` +
-                    `?versionId=${encodeURIComponent(deleteMarkerId)}`,
+                    `?versionId=${deleteMarkerId}`,
```
Comment (electrachong) [inline tests/functional/aws-node-sdk/test/versioning/objectCopy.js]: FYI this should be taken care of by my pr (you may need to rebase once it is merged) : https://github.com/scality/S3/pull/684

**Comment by nicolas2bert** [comment]
@ironman-machine try

File: tests/functional/aws-node-sdk/test/versioning/objectCopy.js [hunk 4]
```diff
@@ -113,7 +116,7 @@ testing('Object Version Copy', () => {
             s3.copyObject(Object.assign({
```
Comment (electrachong) [inline tests/functional/aws-node-sdk/test/versioning/objectCopy.js]: this function is not used anywhere, right? can remove it

File: tests/functional/aws-node-sdk/test/versioning/objectCopy.js [hunk 5]
```diff
@@ -684,7 +687,7 @@ testing('Object Version Copy', () => {
                     Key: sourceObjName, ACL: 'public-read' }, () => {
                     otherAccountS3.copyObject({ Bucket: otherAccountBucket,
                         Key: otherAccountKey,
-                        CopySource: copySource,
+                        CopySource: `${sourceBucketName}/${sourceObjName}`,
```
Comment (electrachong) [inline tests/functional/aws-node-sdk/test/versioning/objectCopy.js]: reason for changing the copy source?

Comment (electrachong) [inline tests/functional/aws-node-sdk/test/versioning/objectCopy.js]: in `                 s3.putObjectAcl({ Bucket: sourceBucketName,
                      Key: sourceObjName, ACL: 'public-read' }`
can you change to 
```
                 s3.putObjectAcl({ Bucket: sourceBucketName,
                      Key: sourceObjName, ACL: 'public-read', VersionId: versionId }
```
?
so you can leave the copy source with the version id

Comment (nicolas2bert) [inline tests/functional/aws-node-sdk/test/versioning/objectCopy.js]: Done :) thanks

**Comment by nicolas2bert** [comment]
@ironman-machine

**Pull request merged**